### DISCUSSION TITLE
Zephyr syscalls aren't getting generated

### DISCUSSION THREAD

**Discussion by chroco** [nasa/fprime#3098] [Q&A] [answered]
I'm trying to get fprime-zephyr-led-blinker working on a stm32 nucleo_f091rc. I experimented on both the main and fprime-v3.5.0 branch and got stuck on the same build error. This error is for atomic_c.h but none of the syscalls get generated.
```
fprime-zephyr-led-blinker/zephyrproject/zephyr/include/zephyr/sys/atomic_c.h:88:10: fatal error: zephyr/syscalls/atomic_c.h: No such file or directory  
   88 | #include <zephyr/syscalls/atomic_c.h>
```
However, it looks to me like syscalls are supposed to be getting generated.
```
build-fprime-automatic-zephyr/zephyr/include/generated/zephyr/autoconf.h:  
#define CONFIG_ATOMIC_OPERATIONS_C 1  
```
Does anyone have any ideas?

**Comment by LeStarch** [comment, marked as answer]
This is likely caused by missing Zephyr settings in `proj.conf` or `Kconfig`.  Note: the community is working on an update zephyr integration.

**Comment by ReggieMarr** [comment]
I believe this is fixed in the latest fprime-zehyr-reference however if anyone is encountering this as I had previously note that rebuilding (not clean) will typically fix it as core issue is just not ordering the dependencies correctly.

Alternatively if you really need to force it you can call the syscall gen task directly and then build (again not clean)
